bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/loader/ScriptFileWatcher.java

```diff
@@ -238,7 +229,9 @@ private String getScriptType(URL url) {
         String fileExtension = fileName.substring(idx + 1);
 
         // ignore known file extensions for "temp" files
-        if (fileExtension.equals("txt") || fileExtension.endsWith("~") || fileExtension.endsWith("swp")) {
+        if (fileExtension.equals("txt") || fileExtension.equals("old") || fileExtension.equals("example")
```

Comment:
Instead of blacklisting wouldn't it be safer to define a whitelist of allowed file extensions?

Otherwise you should put string litterals first in comparisons or may introduce a `Set` of Strings and check if the set contains your blacklisted extension.

Reply:
It would be very difficult to use a whitelist, since any compatible ScriptEngineFactory could be used and there's no way to tell what file extensions might be used.

I'll remove this for now and add a separate PR with a better way to blacklist. I really shouldn't have included this here anyhow, but I've wanted to exclude these extensions for a while.

Reply:
I have reverted this change and will submit another PR for it.